spec/unit/room.spec.ts

```diff
@@ -3534,32 +3534,46 @@ describe("Room", function () {
     });
 
     describe("getLastLiveEvent", () => {
-        let lastEventInMainTimeline: MatrixEvent;
-        let lastEventInThread: MatrixEvent;
-
         it("when there are no events, it should return undefined", () => {
             expect(room.getLastLiveEvent()).toBeUndefined();
         });
 
         it("when there is only an event in the main timeline and there are no threads, it should return the last event from the main timeline", () => {
-            lastEventInMainTimeline = addRoomMainAndThreadMessages(room, 23).mainEvent!;
-            room.addLiveEvents([lastEventInMainTimeline]);
+            const lastEventInMainTimeline = mkMessageInRoom(room, 23);
             expect(room.getLastLiveEvent()).toBe(lastEventInMainTimeline);
         });
 
         it("when there is no event in the room live timeline but in a thread, it should return the last event from the thread", () => {
-            lastEventInThread = addRoomMainAndThreadMessages(room, undefined, 42).threadEvent!;
+            const { thread } = mkThread({ room, length: 0 });
+            const lastEventInThread = mkMessageInThread(thread, 42);
```

Comment:
isn't it impossible for there to be events in the thread and not the main timeline, because the root event is in both?

Reply:
It should also be impossible to have a thread model without events :man_shrugging: (which actually causes this bug).

 I would keep this for completeness and add a comment.